### PR TITLE
feat: add merge workflow

### PR DESCRIPTION
**Pull request by Emnaghz**
# Motivation

This workflow automates the process of building and pushing Docker images for different parts of the project when changes are pushed to the main branch or triggered manually.

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# Checklist:

- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation (Storybook)
- [ ] I have added unit tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: .github/workflows/merge.yml [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+        uses: docker/login-action@v3
+        with:
+          username: ${{ secrets.DOCKERHUB_USERNAME }}
+          password: ${{ secrets.DOCKERHUB_TOKEN }}
+
+      - name: Build and push docker image
+        id: docker_build
+        uses: docker/build-push-action@v6
+        with:
+          context: ./frontend
```
Comment (marrouchi) [inline .github/workflows/merge.yml:52]:
```suggestion
          context: ./
          file: ./frontend/Dockerfile
```

File: .github/workflows/merge.yml [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+        uses: docker/login-action@v3
+        with:
+          username: ${{ secrets.DOCKERHUB_USERNAME }}
+          password: ${{ secrets.DOCKERHUB_TOKEN }}
+
+      - name: Build and push docker image
+        id: docker_build
+        uses: docker/build-push-action@v6
+        with:
+          context: ./widget
```
Comment (marrouchi) [inline .github/workflows/merge.yml:108]:
```suggestion
          context: ./
          file: ./widget/Dockerfile
```

**Review by marrouchi — CHANGES_REQUESTED** [review]
A general remark, I think we can may do the setup of docker and login once to build all the images.

**Comment by Emnaghz** [comment]
In the updated workflow, I've optimized the process by separating the tasks into two distinct jobs:

**Job 1: Path Filtering**
    It filters the paths to determine which parts of the codebase have changed and outputs these results to be used by subsequent jobs.

**Job 2: Build and Push Docker Images**
    This job handles the setup of the building environment, the authentication as well as the conditional builds.

**Explanation:**
When I separated the Docker Hub login process and the environment setup from the build and push steps, I encountered several issues. Each job in GitHub Actions is executed in a separate environment, which means that the authentication status as well as the env setup wasn't shared between other jobs and resulted in many errors.

To address this, I had to ensure that the Docker Hub login and environment setup steps were completed in the same job as the build and push operations.

File: .github/workflows/merge.yml [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+
+      - name: Build and push frontend Docker image
+        if: ${{ needs.paths-filter.outputs.frontend == 'true' }}
+        uses: docker/build-push-action@v6
+        with:
+          context: ./
+          file: ./frontend/Dockerfile
+          platforms: linux/amd64,linux/arm64
+          push: true
+          tags: ${{ secrets.DOCKERHUB_USERNAME }}/hexabot-frontend:latest
```
Comment (marrouchi) [inline .github/workflows/merge.yml:58]:
```suggestion
          tags: hexastack/hexabot-ui:latest
```

File: .github/workflows/merge.yml [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+          file: ./api/Dockerfile
+          platforms: linux/amd64,linux/arm64
+          push: true
+          tags: ${{ secrets.DOCKERHUB_USERNAME }}/hexabot-api:latest
+
+      - name: Build and push widget Docker image
+        if: ${{ needs.paths-filter.outputs.widget == 'true' }}
+        uses: docker/build-push-action@v6
+        with:
+          context: ./
```
Comment (marrouchi) [inline .github/workflows/merge.yml:74]:
```suggestion
          context: ./widget
```
I think we need to set the widget context like this. Can you push and check ?

File: .github/workflows/merge.yml [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+
+      - name: Build and push API Docker image
+        if: ${{ needs.paths-filter.outputs.api == 'true' }}
+        uses: docker/build-push-action@v6
+        with:
+          context: ./
+          file: ./api/Dockerfile
+          platforms: linux/amd64,linux/arm64
+          push: true
+          tags: ${{ secrets.DOCKERHUB_USERNAME }}/hexabot-api:latest
```
Comment (marrouchi) [inline .github/workflows/merge.yml:68]:
```suggestion
          tags: hexastack/hexabot-api:latest
```

**Review by marrouchi — CHANGES_REQUESTED** [review]
I forgot that we need to build the docker image for nlu as well "hexabot-nlu"

**Pull request merged**